src/rules/with.rs

```diff
@@ -49,21 +49,28 @@ pub fn rule(
 
     // /**/
     children.drain_comments(|text| {
+        steps.push_back(crate::builder::Step::Indent);
         steps.push_back(crate::builder::Step::NewLine);
         steps.push_back(crate::builder::Step::Pad);
         steps.push_back(crate::builder::Step::Comment(text));
     });
 
     // expr
+    if let rnix::SyntaxKind::TOKEN_COMMENT =
+        children.peek_prev().unwrap().element.kind()
+    {
+        steps.push_back(crate::builder::Step::NewLine);
+        steps.push_back(crate::builder::Step::Pad);
+    } else {
+        steps.push_back(crate::builder::Step::Whitespace);
+    }
+
     let child = children.get_next().unwrap();
     match layout {
         crate::config::Layout::Tall => {
-            steps.push_back(crate::builder::Step::NewLine);
-            steps.push_back(crate::builder::Step::Pad);
             steps.push_back(crate::builder::Step::FormatWider(child.element));
```

Comment:
Since the with has no limit on how much we can inline, we can get scenarios like this:

```diff
--- a/lib/systems/parse.nix
+++ b/lib/systems/parse.nix
@@ -15,58 +15,52 @@
 # systems that overlap with existing ones and won't notice something amiss.
 #
 { lib }:
-with lib.lists;
-with lib.types;
-with lib.attrsets;
-with lib.strings;
-with (import ./inspect.nix { inherit lib; }).predicates;
-let
+with lib.lists; with lib.types; with lib.attrsets; with lib.strings; with (import ./inspect.nix { inherit lib; })
+.predicates; let
```

A solution could be only allowing one `with` to be inlined, this can be acomplished by checking if the child of the `with` is another `with`; if it is, newline+pad, otherwise inline